pkg/cni/types/types.go

```diff
@@ -0,0 +1,108 @@
+/*
+ Copyright 2023, NVIDIA CORPORATION & AFFILIATES
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containernetworking/cni/pkg/types"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Mellanox/nvidia-k8s-ipam/pkg/cni/k8sclient"
+)
+
+const (
+	DefaultConfDir    = "/etc/cni/net.d/nv-ipam.d"
+	DefaultKubeConfig = DefaultConfDir + "/" + "nv-ipam.kubeconfig"
+	DefaultDataDir    = "/var/lib/cni/nv-ipam"
+	DefaultLogFile    = "/var/log/nv-ipam-cni.log"
+
+	HostLocalDataDir = "state/host-local"
+	K8sNodeNameFile  = "k8s-node-name"
```

Comment:
I don't think we need file for this we can pass it as Env arg

Reply:
how can you pass it ? it is the controller runtime eventually who executes the CNI.

Reply:
I think it will work if we will run it as K8s POD according to this comment 
https://github.com/k8snetworkplumbingwg/multus-cni/blob/b9acfeb6b7fd37c6541327b5b3e1363ab9d57068/pkg/k8sclient/k8sclient.go#L427-L432

Reply:
but multus has a "daemon" mode now. which is why this comment is there, we are not running in this mode.
am i missing anything ?

Reply:
it is not related to the mode. They create the k8s config with thin_enrypoint and after that it run infinite 
https://github.com/k8snetworkplumbingwg/multus-cni/blob/77aac95dfc67ae7662fdab8f406b15330c9b44a6/cmd/thin_entrypoint/main.go 
we can do the the same and add the node name.

Reply:
yes we can :)

but this const definition is still needed in CNI